wled00/FX.h

```diff
@@ -345,25 +344,31 @@ class WS2812FX {
 
   // segment runtime parameters
     typedef struct Segment_runtime { // 28 bytes
-      unsigned long next_time;
-      uint32_t step;
-      uint32_t call;
-      uint16_t aux0;
-      uint16_t aux1;
+      unsigned long next_time;  // millis() of next update
+      uint32_t step;  // custom "step" var
+      uint32_t call;  // call counter
+      uint16_t aux0;  // custom var
+      uint16_t aux1;  // custom var
       byte* data = nullptr;
       bool allocateData(uint16_t len){
         if (data && _dataLen == len) return true; //already allocated
         deallocateData();
         if (WS2812FX::instance->_usedSegmentData + len > MAX_SEGMENT_DATA) return false; //not enough memory
-        data = new (std::nothrow) byte[len];
+        // if possible use SPI RAM on ESP32
```

Comment:
PSRAM is slower than internal RAM theoretically. I don't know whether this makes a difference in practice, even if accessed a lot within the effect function.
Still, not bad to have considering it can be enabled separately with `WLED_USE_PSRAM`.

Reply:
Most people will not use it but I have tested it and it works.

Reply:
Using PSRAM I can see a drop of 5-7 FPS (using Fireworks Starburst effect which uses most memory) with 3-6 segments (600 to 300 pixels each, total of 1800 pixels).